client/src/main/java/io/pravega/client/stream/impl/CheckpointState.java

```diff
@@ -64,23 +64,26 @@
      */
     private final Map<String, Map<Segment, Long>> checkpointPositions;
 
+    private Map<Segment, Long> lastStreamCutPosition;
```

Comment:
This is not being used anywhere.

Reply:
This field will store the positions of the last stream-cut/silent checkpoint so that it does not interfere with `lastCheckpointPosition` where the positions of the actual last checkpoint will be stored. So, instead of having one field store the value of both which was the case before, we have two different fields `lastStreamCutPosition` and `lastCheckpointPosition`.

Reply:
I've removed the field and added two tests to verify the change